components/ShoppingCart/index.js

```diff
@@ -0,0 +1,26 @@
+import BookImage from "../BookImage";
+import BookmarkButton from "../BookmarkButton";
+import DropdownButton from "../DropdownButton";
+import DeleteButton from "../DeleteButton/index.js";
+
+export default function ShoppingCart({ cartItems, handleDelete }) {
+  return (
+    <>
+      <h1>My cart</h1>
+      {cartItems && cartItems.length > 0 ? (
+        cartItems.map((book) => (
+          <div key={book.id}>
+            <BookImage book={book} />
+            <DropdownButton />
+            <DeleteButton book={book} onDelete={handleDelete} />
+            <BookmarkButton />
+          </div>
+        ))
+      ) : (
+        <p>Your cart is empty.</p>
+      )}
```

Comment:
here you'd need to look up the book from the cartItem via the complete book list

```suggestion
      {cartItems && cartItems.length > 0 ? (
        cartItems.map((cartItem) => {
          const book = books.find((book) => book.id === cartItem.bookId);
          if (!book) {
            return null;
          }
          return (
            <div key={book.id}>
              <BookImage book={book} />
              <DropdownButton />
              <DeleteButton book={book} onDelete={handleDelete} />
              <BookmarkButton />
            </div>
          );
        })
      ) : (
        <p>Your cart is empty.</p>
      )}
```